gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/ProjectService.java

```diff
@@ -40,5 +46,102 @@ public Page<Project> getAll(ProjectCriteria criteria, Pageable pageable) throws
         return projectRepository.findAll(ProjectPredicates.build(criteria), pageable);
     }
 
+    public Project createProject(ProjectCriteria criteria, Project project) throws Exception {
+        Instant now = Instant.now();
+
+        if (project.getId() != null) {
+            throw new GendoxException("NEW_PROJECT_ID_IS_NOT_NULL", "Project id must be null", HttpStatus.BAD_REQUEST);
+        }
+
+        // Validate the criteria here based on your requirements
+        Predicate predicate = ProjectPredicates.build(criteria);
+
+        // Check if the criteria matches any existing projects
+        boolean criteriaMatchesExistingProjects = projectRepository.exists(predicate);
+
+        if (!criteriaMatchesExistingProjects) {
+            throw new GendoxException("PROJECT_CRITERIA_DOES_NOT_MATCH_EXISTING", "The criteria do not match any existing projects", HttpStatus.BAD_REQUEST);
+        }
```

Comment:
I dont see any reason to need a unique project (by any criteria). You can delete this, or please explain if i miss something.